### PR TITLE
Use credential fetcher to fetch token for XDS

### PR DESCRIPTION
**Pull request by JimmyCYJ**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
lgtm

File: security/pkg/nodeagent/caclient/credentials.go [hunk 1]
```diff
@@ -132,25 +132,36 @@ func (t *TokenProvider) GetTokenForXDS() (string, error) {
...
-	if err != nil {
-		log.Warnf("failed to fetch token from file: %v", err)
-		return "", nil
+	var tok string
+	var err error
+	if t.opts.CredFetcher != nil {
+		// When running at a non-k8s platform, use CredFetcher to get credential.
+		tok, err = t.opts.CredFetcher.GetPlatformCredential()
+		if err != nil {
+			return "", fmt.Errorf("fetch platform credential: %v", err)
```
Comment (JimmyCYJ) [inline security/pkg/nodeagent/caclient/credentials.go:141]: Not clear if this should return nil and log.Warnf(), I notice that we don't return err in this call.

Comment (JimmyCYJ) [reply security/pkg/nodeagent/caclient/credentials.go:141]: Ah, ignore this message, we do return err.

**Review by lei-tang — APPROVED** [review]

File: security/pkg/nodeagent/caclient/credentials.go [hunk 2]
```diff
@@ -132,25 +132,37 @@ func (t *TokenProvider) GetTokenForXDS() (string, error) {
...
+		}
+		tok = string(tokBytes)
 	}
 	// For XDS flow, the token exchange is different from that of the CA flow.
 	if t.opts.TokenManager == nil {
 		return "", fmt.Errorf("XDS token exchange is enabled but token manager is nil")
 	}
-	if strings.TrimSpace(string(tok)) == "" {
+	if strings.TrimSpace(tok) == "" {
 		return "", fmt.Errorf("the JWT token for XDS token exchange is empty")
```
Comment (liminw) [inline security/pkg/nodeagent/caclient/credentials.go:160]: NIT: JWT token -> token. This may not be a JWT.

**Review by liminw — COMMENTED** [review]
LGTM.

File: security/pkg/nodeagent/caclient/credentials.go (hunk 2 above)
Comment (JimmyCYJ) [reply security/pkg/nodeagent/caclient/credentials.go]: Done. Thanks.

**Pull request merged**